File: src/main/java/app/domain/equipment/Equipment.java
package app.domain.equipment;

import javax.persistence.*;

@Entity
@Table(name = "equipment")
@IdClass(EquipmentId.class)
public class Equipment {

    @Id
    @Column
    private int id;

    @Id
    @Column
    private int slotNum;

    @Column
    private int itemId;

    public Equipment() {

    }

    public int getId() {
        return id;
    }

    public int getSlotNum() {
        return slotNum;
    }

    public int getItemId() {
        return itemId;
    }

}

File: src/frontend/src/models/equipment.js
import { hasPropertyOrDefault } from "@/utils/hasPropertyOrDefault";

export default class Equipment {
    constructor(config) {
        this._id = hasPropertyOrDefault(config, "id", "");
        this._slotNum = hasPropertyOrDefault(config, "slotNum", "");
        this._itemId = hasPropertyOrDefault(config, "itemId", "");
    }


    get id() {
        return this._id;
    }

    get slotNum() {
        return this._slotNum;
    }

    get itemId() {
        return this._itemId;
    }
}